server/controllers/list.js

```diff
@@ -22,28 +21,51 @@ const {
 const { updateSubdocumentFields } = require('../common/utils/helpers');
 
 const createList = (req, resp) => {
-  const { description, isListPrivate, name, userId, cohortId } = req.body;
+  const { description, isListPrivate, name, cohortId } = req.body;
+  const {
+    user: { _id: userId }
+  } = req;
 
   const list = new List({
     cohortId,
     description,
     isPrivate: isListPrivate,
+    memberIds: userId,
     name,
-    ownerIds: userId
+    ownerIds: userId,
+    viewersIds: userId
   });
 
-  list.save((err, doc) => {
-    if (err) {
-      return resp
-        .status(400)
-        .send({ message: 'List not saved. Please try again.' });
-    }
+  if (cohortId && !isListPrivate) {
+    Cohort.findOne({ _id: cohortId })
+      .then(cohort => {
+        const { memberIds } = cohort;
+        list.viewersIds = memberIds;
```

Comment:
what if user isn't a cohort's member? Should we check here something like this:
```
if(memberIds.includes(userId)){
throw new BadRequestException();
}
```

Reply:
when creating a list, he always gonna be viewer member and owner by default.
When creating a list within cohort he is already a cohort member. 
Otherwise, how he can create list within cohort??

Reply:
Hmm, probably you're right but if he will have the session sid (cookie) and cohortId he could do this via postman